File: CreationClasses/metier/etudiant.py
from metier.personne import Personne

class Etudiant (Personne):

    def __init__(self, nom: str ="Inconnu", semestre: float = 1) -> None:
        super().__init__(nom)
        self.__semestre = semestre

    def afficher(self) -> str:
        return self.get_nom() + " est dans le semestre " + str(self.__semestre)

File: CreationClasses-Java/src/domaine/Etudiant.java
package domaine;

public class Etudiant extends Personne {

    private int semestre;

    public Etudiant() {
//        super();
        semestre = 1;
    }

    public Etudiant(String nom) {
        this(nom, 1);
    }

    public Etudiant(String nom, int semestre) {
        super(nom);
        this.semestre = semestre;
    }

    @Override
    public String afficher() {
        return getNom() + " est dans le semestre " + semestre;
    }
}
